### PR TITLE
amp-carousel: Remove BaseSlides

### PR DESCRIPTION
**Pull request by samouri**
### summary
**Context**: I plan on creating a `buildDom` for amp-carousel.

`BaseSlides` was designed for extension but only has a single child, which IMHO ends up just making the already complex code slightly more complex by forcing you to understand the sequence of overrides, build calls, etc.

**before**
```
               base-slides — slidescroll
              / 
base-carousel
              \
                scrollable-carousel
```

**after**
```
                slidescroll
              / 
base-carousel
              \
                scrollable-carousel
```

### CODE REVIEW

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 1]
```diff
@@ -927,4 +1019,104 @@ export class AmpSlideScroll extends BaseSlides {
...
+    } else {
+      this.clearAutoplay_();
+    }
+  }
+
+  /**
+   * Clear the autoplay timer.
+   * @private
+   */
+  clearAutoplay_() {
```
Comment (caroqliu) [inline extensions/amp-carousel/0.1/slidescroll.js:1099]: nit: it's not obvious the difference between `clearAutoplay` and `removeAutoplay`, can we rename?
```suggestion
  clearAutoplayTimer_() {
```

**Review by caroqliu — APPROVED** [review]
nice! thanks for doing this! just had one suggestion

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 1 above)
Comment (samouri) [reply extensions/amp-carousel/0.1/slidescroll.js:1099]: It still isn't obvious to me....why are the timer and the attributes handled in separate funcs?

Comment (caroqliu) [reply extensions/amp-carousel/0.1/slidescroll.js]: I'm not sure either. `clearAutoplay` appears to be called much more frequently than `removeAutoplay`...

**Pull request merged**